### ISSUE TITLE
Add support for OneDrive 'business' accounts

### ISSUE THREAD

**Issue by michaelrw10** [Kyrodan/KeeAnywhere#28]
There is a distinction between personal onedrive accounts and 'business' onedrive accounts that you get from work/university/organization. 

Because microsoft changed their sync client not too long ago, users sometimes run into this problem when trying to log in with a business account. For example, when I had to get the sync client setup on my personal PC, I had to use the troubleshooting info in these articles to be able to log in successfully. This should give you enough info about the issue, if you don't already know. 

[Get started with the OneDrive for Business Next Generation Sync Client in Windows](https://support.office.com/en-us/article/Get-started-with-the-OneDrive-for-Business-Next-Generation-Sync-Client-in-Windows-615391c4-2bd3-4aae-a42a-858262e42a49?ui=en-US&rs=en-US&ad=US)

[Deploying the OneDrive for Business Next Generation Sync Client in an enterprise environment](https://support.office.com/en-us/article/Deploying-the-OneDrive-for-Business-Next-Generation-Sync-Client-in-an-enterprise-environment-3f3a511c-30c6-404a-98bf-76f95c519668)

[Administrative settings for the OneDrive for Business Next Generation Sync Client](https://support.office.com/en-us/article/Administrative-settings-for-the-OneDrive-for-Business-Next-Generation-Sync-Client-0ecb2cf5-8882-42b3-a6e9-be6bda30899c)

Thanks for the help! I would really appreciate if business accounts could be added with your plugin!

**Comment by Kyrodan** [comment]
OneDrive Business (in context of Office365 and maybe in Sharepoint Online) needs a payed developer account to get an app key and to operate an app. Therefore this is a blacklistet feature like stated in https://github.com/Kyrodan/KeeAnywhere/wiki/Contributing#blacklisted-features beacuse I don't wan't to run into a "pay dependency" for such an account.

Additionally Sharepoint *on Premise* is more complex to implement and configuration goes beyond the scope of KeeAnywhere's key features (simplicity).

Sponsoring a developer account for at least 3 years in advance could be a reasonable compromise.

**Comment by michaelrw10** [comment]
Understood. Although I'm a little confused why someone would have to sponsor (i.e. Pay for) a developers account for 3 years. Are you suggesting it would take 3 years to develop? 

It's moot, however, because I'm not in a position to sponsor anything.. But I am slightly curious. 

At any rate, thx for the info

**Comment by Kyrodan** [comment]
>  Although I'm a little confused why someone would have to sponsor (i.e. Pay for) a developers account for 3 years. Are you suggesting it would take 3 years to develop? 

No, it would not take three years to develop. But a developer account is also needed for the whole time KeeAnywhere should be able to access OneDrive Business. With the three years "sponsoring" of a developer account I will make sure that KeeAnywhere would work with OneDrive Business for at least three years: I have to register KeeAnywhere as an app for OneDrive Business and this app key is only valid as long as I pay my developer account. So if I quit my developer account KeeAnywhere won't work with OneDrive Business anymore. And that would be annoying for users that have a Business account.

I also want to clarify my opinion regarding "business services" in general: 
I'm develop this plugin in my spare time and I don't use most of the the providers by myself. I just do it because it's fun for me. I'm happy about every user that finds this plugin useful regardless whether he uses this plugin in a commercial environment or not. I don't want to earn any money with this plugin. But I also don't want to be a charge-free slave in situation where a commercial usage with a business value is clearly given. And I also want to pay fees for the ability that others can use this plugin for free. Everyone who has a OneDrive Business account pays for it (when not now, than later) and Microsoft clearly states OneDrive Business as a commercial product someone has to pay for. All payable Office 365 products for private (non-commercial) users do not contain a OneDrive Business account.

Please don't take the following words personally, it's not an attack agains you @michaelrw10. But I want to make others think about the honoration of others work and the spirit of the open source community:
> It's moot, however, because I'm not in a position to sponsor anything

That's the answer I got everytime. If you (or your employer) finds something useful and it has a business value, please try to "sponsor" it (again: it's *not* "to pay for sth.", because I dont' want to earn money with that). Three years of a developer account costs less than one fourth of a contractor day in Frankfurt Area - so it's not much. And it's an "invest" in the community.

**Comment by chrisg95** [comment]
Hello Kyrodan,

Thanks for Keeanywhere, i use it (and love it) for my personnal keepass database and now i will be happy to use it with my professionnal database.

I am ready to invest in the community by sponsoring the developpement of this functionnality, how can we start ?

Regards,

Chris

**Comment by michaelrw10** [comment]
This is great news! I hope that this is still an option. Looking forward to finding out. 

- Michael

**Comment by Kyrodan** [comment]
Hi @chrisg95,

I wrote you an email some days ago and didn't received a reply so far. Did you get it? Maybe it gets lost in your spam folder?

Regards,
Daniel

**Comment by kribjo** [comment]
I am also very interested in this. It seems it is a one-time only fee? Would an individual account suffice? Which region would it be for? Is this still an option?
https://docs.microsoft.com/sv-se/windows/uwp/publish/account-types-locations-and-fees

**Comment by sbourdette** [comment]
I would be interested in this feature

**Comment by firecreeper** [comment]
would be great if you can adopt the features from KeePassOneDriveSync because i don't want to manage different plugins for the same issue
